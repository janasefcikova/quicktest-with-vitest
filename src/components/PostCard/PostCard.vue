<template>
    <div>
        <div v-if="post" style="padding: 1em; border: 2px solid gray; border-radius: 5px;">
            <h1>Post card without shadcn</h1>
            <h2 data-testid="post-title">{{ post.title }}</h2>
            <p data-testid="post-body">{{ post.body }}</p>
        </div>
        <p v-if="loading" data-testid="loader">Loading...</p>
        <p v-if="error" data-testid="error-message">{{ error }}</p>
    </div>
</template>

<script setup>

import { ref, onMounted } from "vue";
import axios from "axios";

const post = ref(null);
const loading = ref(true);
const error = ref(null);

const fetchPost = async () => {
    try {
        const { data } = await axios.get(
            "https://jsonplaceholder.typicode.com/posts/1"
        );
        post.value = data;
    } catch (err) {
        error.value = err.message;
    } finally {
        loading.value = false;
    }
};

onMounted(() => {
    fetchPost();
});

</script>