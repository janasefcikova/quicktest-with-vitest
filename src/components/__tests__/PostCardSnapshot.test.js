/* eslint-disable no-undef */
import axios from 'axios'
import PostCard from '../PostCard.vue'
import { mount, flushPromises } from '@vue/test-utils'

const mockPost = {
  userId: 1,
  id: 1,
  title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
  body: 'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'
}

describe('Post Card Component', () => {
  test('before fetching data, post card show loading', async () => {
    vi.spyOn(axios, 'get').mockResolvedValueOnce({ data: mockPost })
    const wrapper = mount(PostCard)
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('with valid data', async () => {
    vi.spyOn(axios, 'get').mockResolvedValueOnce({ data: mockPost })
    const wrapper = mount(PostCard)
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('displays an error message if fetching a post fails', async () => {
    vi.spyOn(axios, 'get').mockRejectedValueOnce(new Error('Error occurred'))
    const wrapper = mount(PostCard)
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})