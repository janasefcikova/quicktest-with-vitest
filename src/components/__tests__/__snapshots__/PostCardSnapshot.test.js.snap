// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post Card Component > before fetching data, post card show loading 1`] = `
"<div>
  <!--v-if-->
  <p data-testid="loader">Loading...</p>
  <!--v-if-->
</div>"
`;

exports[`Post Card Component > displays an error message if fetching a post fails 1`] = `
"<div>
  <!--v-if-->
  <!--v-if-->
  <p data-testid="error-message">Error occurred</p>
</div>"
`;

exports[`Post Card Component > with valid data 1`] = `
"<div>
  <div>
    <h1 data-testid="post-title">sunt aut facere repellat provident occaecati excepturi optio reprehenderit</h1>
    <p data-testid="post-body">quia et suscipit
      suscipit recusandae consequuntur expedita et cum
      reprehenderit molestiae ut ut quas totam
      nostrum rerum est autem sunt rem eveniet architecto</p>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
